### PR TITLE
code review for sdk6.5.x

### PR DESCRIPTION
**Pull request by HuFangshuai**
this PR has been created for banner function code review

### CODE REVIEW

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 1]
```diff
@@ -117,30 +112,40 @@ public void run() {
...
+        AdConfig.AdSize adSizeType = null;
         int adWidthInDp = localExtras.containsKey(KEY_AD_WIDTH) ? (int)localExtras.get(KEY_AD_WIDTH) : 0;
         int adHeightInDp = localExtras.containsKey(KEY_AD_HEIGHT) ? (int)localExtras.get(KEY_AD_HEIGHT) : 0;
 
         if((adWidthInDp == 300 && adHeightInDp == 250) || (adWidthInDp == 336 && adHeightInDp == 280)) {
-            return AdConfig.AdSize.VUNGLE_MREC;
-        } else {
-            return null;
+            adSizeType = AdConfig.AdSize.VUNGLE_MREC;
+        } else if (adWidthInDp == 300 && adHeightInDp == 50) {
```
Comment (alexander-gubov) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: Have this values in AdSize enum

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 2]
```diff
@@ -117,30 +112,40 @@ public void run() {
             return;
         }
 
-        //currently we only support MREC for banners. This will require to be reworked once we add other sizes
-        if(adConfig.getAdSize() != AdConfig.AdSize.VUNGLE_MREC) {
+        if(mAdSize != AdConfig.AdSize.VUNGLE_MREC && mAdSize != AdConfig.AdSize.BANNER_SHORT
+           && mAdSize != AdConfig.AdSize.BANNER && mAdSize != AdConfig.AdSize.BANNER_LEADERBOARD) {
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: How about this? 

```
if(!isValidBannerSize (mAdSize)) {
    post unsupported banner size error
}

private boolean isValidBannerSize(AdConfig.AdSize mAdSize) {
      return (mAdSize == AdConfig.AdSize.VUNGLE_MREC || 
                 mAdSize == AdConfig.AdSize.BANNER_SHORT || 
                 mAdSize == AdConfig.AdSize.BANNER || 
                 mAdSize == AdConfig.AdSize.BANNER_LEADERBOARD);
}
```

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 3]
```diff
@@ -219,8 +224,8 @@ public void onAdStart(@NonNull String placementReferenceId) {
                     mCustomEventBannerListener.onBannerImpression();
                 }
 
-                //Let's load it again to mimic auto-cache
-                sVungleRouter.loadAdForPlacement(mPlacementId, mVungleRouterListener);
+                //et's load it again to mimic auto-cache
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: COSMETIC: [L]et's load it again to mimic auto-cache

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 4]
```diff
@@ -264,15 +269,12 @@ protected void onVisibilityChanged(@NonNull View changedView, int visibility) {
                                         }
                                     }
                                 };
-                                vungleBannerAd = sVungleRouter.getVungleBannerAd(placementReferenceId, adConfig);
+                                vungleBannerAd = sVungleRouter.getVungleBannerAd(placementReferenceId, mAdSize);
                                 if(vungleBannerAd != null) {
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: `isSuccess` is a redundant flag. Remove it.

How about? 

```
vungleBannerAd = sVungleRouter.getVungleBannerAd(...)
if(vungleBannerAd != null) {
           addView(vungleBannerAd)
} else {
           NO FILL Error
}
```

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java [hunk 5]
```diff
@@ -264,15 +269,12 @@ protected void onVisibilityChanged(@NonNull View changedView, int visibility) {
...
-                                        mCustomEventBannerListener.onBannerLoaded(layout);
-                                        MoPubLog.log(LOAD_SUCCESS, ADAPTER_NAME);
-                                    }
+                                    isSuccess = true;
+                                    layout.addView(vungleBannerAd);
+                                    mCustomEventBannerListener.onBannerLoaded(layout);
+                                    MoPubLog.log(LOAD_SUCCESS, ADAPTER_NAME);
                                 }
                                 if(!isSuccess) {
                                     mHandler.post(new Runnable() {
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:324]: - From which thread this is called? 
- addView() seems to be only candidate to be called on the main UI thread.
- Are onBannerLoaded() and log() calls are synchronous calls?

File: Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java [hunk 6]
```diff
@@ -418,7 +418,7 @@ private VungleMediationSettings(@NonNull final Builder builder) {
             this.isSoundEnabled = builder.isSoundEnabled;
             this.flexViewCloseTimeInSec = builder.flexViewCloseTimeInSec;
             this.ordinalViewCount = builder.ordinalViewCount;
-            this.autoRotateEnabled = builder.autoRotateEnabled;
+            this.adOrientation = builder.adOrientation;
         }
     }
 }
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java]: EOF could be issue for the code scanning tools such as static analysis.

File: Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java [hunk 7]
```diff
@@ -164,29 +172,55 @@ public void loadAdForPlacement(String placementId, VungleRouterListener routerLi
         }
     }
 
+    void loadBannerAd(String placementId, AdConfig.AdSize adSize, VungleRouterListener routerListener) {
+        switch (sInitState) {
+            case NOTINITIALIZED:
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, "loadBannerAdForPlacement is called before " +
+                        "initialization starts. This is not an expect case.");
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java]: How about? 

`can't display banner before the Vungle adapter initialization` OR
`Please initialize the Vungle adapter before loading the banner` OR
`loadBannerAdForPlacement is called before the adapter initialization!!!`

File: Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java [hunk 8]
```diff
@@ -164,29 +172,55 @@ public void loadAdForPlacement(String placementId, VungleRouterListener routerLi
         }
     }
 
+    void loadBannerAd(String placementId, AdConfig.AdSize adSize, VungleRouterListener routerListener) {
```
Comment (devendravaja-vungle) [inline Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java]: Is it mandatory to pass the routerListener? `If yes` than we should change the signature to something like 

`void loadBannerAd(@NonNull String placementId, AdConfig.AdSize adSize, @NonNull VungleRouterListener routerListener)` 

Check other functions and add the annotations if required.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 1 above)
Comment (devendravaja-vungle) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: or separate IntDef

**Review by devendravaja-vungle — COMMENTED** [review]
The comments are solely based on the abstract view of the code. Logic and memory management may

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 1 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: fixed

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 4 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: fixed

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 5 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java:324]: these inner code part running in mHandler and mHandler instance using MainLooper, so addView() here is ok.

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 2 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: fixed

File: Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java (hunk 3 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java]: fixed

File: Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java (hunk 6 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleRewardedVideo.java]: fixed

File: Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java (hunk 7 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java]: fixed

File: Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java (hunk 8 above)
Comment (HuFangshuai) [reply Vungle/src/main/java/com/mopub/mobileads/VungleRouter.java]: fixed

**Review by devendravaja-vungle — COMMENTED** [review]
Seems good to me 👍

**Review by palrahul — APPROVED** [review]
lgtm

**Pull request merged**